File: src/main/java/com/wafflestudio/draft/dto/response/PreferenceInRegionResponse.java
package com.wafflestudio.draft.dto.response;

import com.wafflestudio.draft.model.Preference;
import com.wafflestudio.draft.model.Region;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class PreferenceInRegionResponse {
    private Long id;
    private String name;
    private String depth1;
    private String depth2;
    private String depth3;
    private List<PreferenceResponse> preferences;

    public PreferenceInRegionResponse(Region region, List<Preference> preferences) {
        this.id = region.getId();
        this.name = region.getName();
        this.depth1 = region.getDepth1();
        this.depth2 = region.getDepth2();
        this.depth3 = region.getDepth3();
        List<PreferenceResponse> preferenceResponses = new ArrayList<>();
        for (Preference preference : preferences) {
            preferenceResponses.add(new PreferenceResponse(preference));
        }
        this.preferences = preferenceResponses;
    }
}

File: src/main/kotlin/com/wafflestudio/draft/dto/response/PreferenceInRegionResponse.kt
package com.wafflestudio.draft.dto.response

import com.wafflestudio.draft.dto.PreferenceDTO
import com.wafflestudio.draft.model.Preference
import com.wafflestudio.draft.model.Region

class PreferenceInRegionResponse(
        var id: Long? = null,
        var name: String? = null,
        var depth1: String? = null,
        var depth2: String? = null,
        var depth3: String? = null,
        var preferences: List<PreferenceDTO.Response>? = null
) {
    constructor(region: Region, preferences: List<Preference?>?) : this(
            region.id,
            region.name,
            region.depth1,
            region.depth2,
            region.depth3,
            preferences?.map { PreferenceDTO.Response(it) }
    )
}
